### PR TITLE
POC - native VEC support for Target - postMessage impl

### PR DESCRIPTION
**Pull request by dcottingham**
This is POC to show how Alloy can support loading external scripts using postMessage when run in an iframe. This is necessary in order to support the Target visual experience composer (VEC). More details here:
https://wiki.corp.adobe.com/pages/viewpage.action?spaceKey=Target&title=Alloy+VEC+Integration

* Once the design is agreed upon I will create test cases.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

File: src/core/preloadHook.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    if (
+      event.data.type !== "preloadHook" ||
+      !ALLOW_LIST.includes(event.origin)
+    ) {
+      return;
+    }
+
+    try {
+      // eslint-disable-next-line no-eval
+      window.eval(event.data.value);
```
Comment (jonsnyder) [inline src/core/preloadHook.js:24]: This is a big red flag for any security scanner.

Comment (dcottingham) [reply src/core/preloadHook.js:24]: So there are two overall goals I'm trying to achieve - outlined in this wiki I put together:  https://wiki.corp.adobe.com/pages/viewpage.action?spaceKey=Target&title=Alloy+VEC+Integration

The SPA support - which is what uses the monitoring hooks - does use postMessage to send those events to the VEC in the parent window.  That will work fine without any explicit changes to Alloy.  

This PR addresses the second goal - native VEC support.  So basically we need some means of executing our target.js script inside of the iframe before Alloy instances are created.  Given the constraints, I can't think of any way to implement this where Alloy doesn't use eval to run the script.

Comment (dcottingham) [reply src/core/preloadHook.js]: I've updated this to use the reactor-load-script instead of eval

Comment (jonsnyder) [reply src/core/preloadHook.js]: Wasn't one of the goals not requiring the customer to allow scripts loaded from target?

**Comment by jfkhoury** [comment]
@dcottingham Can we close this PR at this point?

**Comment by dcottingham** [comment]
Going with browser extension instead of script injection

**Pull request closed without merging**